atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java

```diff
@@ -65,75 +66,59 @@ public SSLSocketFactory load(InetSocketAddress host) throws Exception {
-    private final String keyspace;
-    private final Optional<CassandraCredentialsConfig> creds;
-    private final boolean isSsl;
-    private final int socketTimeoutMillis;
-    private final int socketQueryTimeoutMillis;
-
-    public CassandraClientFactory(InetSocketAddress addr,
-                                  String keyspace,
-                                  Optional<CassandraCredentialsConfig> creds,
-                                  boolean isSsl,
-                                  int socketTimeoutMillis,
-                                  int socketQueryTimeoutMillis) {
+    private final CassandraKeyValueServiceConfig config;
+
+    public CassandraClientFactory(InetSocketAddress addr, CassandraKeyValueServiceConfig config) {
         this.addr = addr;
-        this.keyspace = keyspace;
-        this.creds = creds;
-        this.isSsl = isSsl;
-        this.socketTimeoutMillis = socketTimeoutMillis;
-        this.socketQueryTimeoutMillis = socketQueryTimeoutMillis;
+        this.config = config;
     }
 
     @Override
     public Client create() throws Exception {
         try {
-            return getClient(addr, keyspace, creds, isSsl, socketTimeoutMillis, socketQueryTimeoutMillis);
+            return getClient(addr, config);
         } catch (Exception e) {
-            String message = String.format("Failed to construct client for %s/%s", addr, keyspace);
-            if (isSsl) {
+            String message = String.format("Failed to construct client for %s/%s", addr, config.keyspace());
+            if (config.usingSsl()) {
                 message += " over SSL";
             }
             throw new ClientCreationFailedException(message, e);
         }
     }
 
     private static Cassandra.Client getClient(InetSocketAddress addr,
-                                              String keyspace,
-                                              Optional<CassandraCredentialsConfig> creds,
-                                              boolean isSsl,
-                                              int socketTimeoutMillis,
-                                              int socketQueryTimeoutMillis) throws Exception {
-        Client ret = getClientInternal(addr, creds, isSsl, socketTimeoutMillis, socketQueryTimeoutMillis);;
+                                              CassandraKeyValueServiceConfig config) throws Exception {
+        Client ret = getClientInternal(addr, config);
         try {
-            ret.set_keyspace(keyspace);
-            log.debug("Created new client for {}/{} {} {}", addr, keyspace, (isSsl ? "over SSL" : ""),
-                    creds.isPresent() ? " as user " + creds.get().username() : "");
+            ret.set_keyspace(config.keyspace());
+            log.debug("Created new client for {}/{} {} {}", addr, config.keyspace(), (config.usingSsl() ? "over SSL" : ""),
+                    config.credentials().isPresent() ? " as user " + config.credentials().get().username() : "");
```

Comment:
nit: when `config.usingSsl() == true` and `config.credentials().isPresent()` there will be two spaces after SSL in the rendered string: "... over SSL[ ][ ]as user bob".  To fix, change "[ ]as user[ ]" to "as user[ ]"